src/pages/oncall-me/oncall-me.ts

```diff
@@ -0,0 +1,94 @@
+import { Component } from '@angular/core';
+import { IonicPage, NavController, NavParams, ActionSheetController, ToastController } from 'ionic-angular';
+import { IrisProvider, OncallUser} from '../../providers/iris/iris';
+import { OncallTeamPage } from '../oncall-team/oncall-team';
+import { IrisInfoProvider } from '../../providers/iris_info/iris_info';
+import { LogoutProvider } from '../../providers/logout/logout';
+/**
+ * Generated class for the OncallMePage page.
+ *
+ * See https://ionicframework.com/docs/components/#navigation for more info on
+ * Ionic pages and navigation.
+ */
+
+@Component({
+  selector: 'page-oncall-me',
+  templateUrl: 'oncall-me.html',
+})
+export class OncallMePage {
+  user: OncallUser;
+  loading: boolean;
+  loadingError: boolean;
+
+  constructor(private logOut: LogoutProvider, public navCtrl: NavController, public navParams: NavParams, private actionCtrl: ActionSheetController, public iris: IrisProvider, private toastCtrl: ToastController, private irisInfo: IrisInfoProvider) {
+  }
+  
+  ionViewWillEnter() {
+    this.getUser();  
+  }
+
+  getUser(){
+    this.user = new OncallUser;
+    this.loading = true;
+    this.loadingError = false;
+    this.iris.getOncallUser(this.irisInfo.username).subscribe(
```

Comment:
This is because getOncallUser() calls 3 different Oncall api endpoints to get all the info i needs so it doesn't have the structure of a OncallUser object it's actually a Object[] where each element in the array is a response from a different api call

``